### PR TITLE
Publish the failing test's name and class along with the integration test results

### PR DESCRIPTION
**Pull request by prudhvigodithi**
### Description
Publish the failing test's name and class along with the integration test results.

### Issues Resolved
Part of https://github.com/opensearch-project/opensearch-metrics/issues/68 and https://github.com/opensearch-project/opensearch-build-libraries/issues/535.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by peterzhuamazon — APPROVED** [review]
Thanks!

File: vars/publishIntegTestResults.groovy [hunk 1]
```diff
@@ -88,8 +97,159 @@ void call(Map args = [:]) {
...
+                            else
+                                echo "Failed to update alias. Error message: \$update_alias_response"
+                            fi
+                        else
+                            echo "Failed to create index. Error message: \$create_index_response"
+                            exit 1
+                        fi
+                    fi
+                    if [ -s ${testFailuresFile} ]; then
+                        echo "File Exists, indexing results."
```
Comment (gaiksaya) [inline vars/publishIntegTestResults.groovy:243]: nit:```suggestion
                        echo "File Exists, indexing failed tests."
```

File: vars/publishIntegTestResults.groovy [hunk 2]
```diff
@@ -224,6 +397,34 @@ void indexFailedTestData(indexName, testRecordsFile) {
     }
 }
 
+def generatefailedTestJson(componentName, componentRepo, componentRepoUrl, version,
```
Comment (gaiksaya) [inline vars/publishIntegTestResults.groovy:400]: nit:

```suggestion
def generateFailedTestJson(componentName, componentRepo, componentRepoUrl, version, 
```

**Comment by prudhvigodithi** [comment]
Thanks @peterzhuamazon and @gaiksaya, I have updated the requested changes in my latest commit.

File: vars/publishIntegTestResults.groovy [hunk 3]
```diff
@@ -88,8 +97,159 @@ void call(Map args = [:]) {
...
+            break
+        case failedTests.contains("No Failed Test"):
+            break
+        default:
+            failedTests.collect { failedTest ->
+                def match = failedTest.split("#")
+                if (match) {
+                    def testResultJsonContent = generateFailedTestJson(componentName, componentRepo, componentRepoUrl, version, integTestBuildNumber, 
+                        integTestBuildUrl, distributionBuildNumber, distributionBuildUrl, buildStartTime, rc, rcNumber, 
+                        platform, architecture, distribution, componentCategory, securityType, match[0].trim(), match[1].trim())
```
Comment (gaiksaya) [inline vars/publishIntegTestResults.groovy:122]: Did not get this part. How are we splitting based on #?

Comment (prudhvigodithi) [reply vars/publishIntegTestResults.groovy:122]: Ya this for splitting `test_class#test_name`, I have added a small test for both OpenSearch and Dashboards.

**Comment by prudhvigodithi** [comment]
> Looks like some groovy tests are failing due to recent commit. Please fix before merging.
> Thanks!

Ya I forgot to update the tests when changed to sentence `indexing failed tests`, I'm testing the entire `sh` command and its output. Just pushed it in my latest commit.

**Comment by prudhvigodithi** [comment]
Merging this as the Mend Base branch commit: [4e94ed7991c7192faa94996988c1deacb8b9ec98](https://github.com/opensearch-project/opensearch-build-libraries/runs/31430960020), coming from https://github.com/opensearch-project/opensearch-build-libraries/pull/531. @gaiksaya @peterzhuamazon

**Pull request merged**